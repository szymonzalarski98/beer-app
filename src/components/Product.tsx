import React from 'react';
import { TouchableOpacity, Image, View, Text, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { deviceWidth } from '../styles/helpers.style';
import Property from './Property';

// name, image, tags for filter
interface Props {
  id: number,
  name: String;
  image: String;
  abv: number,
  ibu: number,
};

const Product: React.SFC<Props> = (props) => {
  const { name, image, abv, ibu, id } = props;
  const navigation = useNavigation();
  return (
    <TouchableOpacity testID="product" onPress={() => navigation.navigate('Details', { id })} style={styles.container}>
      {
        image !== null
        ?
        <Image source={{ uri: image }} style={styles.image} resizeMode={'contain'} />
        :
        <View style={styles.blankImage} />
      }
      <View style={styles.infoContainer}>
        <Property title="" text={name} />
        <Property title="Alcohol by volume: " text={abv} />
        <Property title="Ibu: " text={ibu} />
      </View>
    </TouchableOpacity>
  );
};

export default Product;

const styles =  StyleSheet.create({
  container: {
    flex: 1,
    height: 100,
    backgroundColor: '#fff',
    alignItems: 'center',
    paddingHorizontal: deviceWidth * 0.05,
    paddingVertical: 10,
    flexDirection: 'row',
  },
  infoContainer: {
    flexDirection: 'column',
    justifyContent: 'center',
    flex: 1,
    marginLeft: deviceWidth * 0.05,
  },
  image: {
    aspectRatio: 1,
    width: 80,
    height: 80,
  },
  blankImage: {
    width: 80,
    height: 80,
    borderRadius: 5,
    backgroundColor: '#fafafa'
  }
});