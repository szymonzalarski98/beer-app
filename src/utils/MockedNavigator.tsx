import React, { ReactNode } from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
interface Props {
  component: ReactNode,
  params?: {}
}

const Stack = createStackNavigator();
const MockedNavigator: React.SFC<Props> = ({component, params = {}}) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="MockedScreen"
          component={component}
          initialParams={params}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default MockedNavigator;