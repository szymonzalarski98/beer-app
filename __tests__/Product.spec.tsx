import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import MockedNavigator from '../src/utils/MockedNavigator';
import Product from '../src/components/Product';

describe('Test Product component', () => {
  const ProductComponent = () => <Product id={1} name="Buzz" image="test-image-link" abv={4.5} ibu={4.1} />;
  it ('Should render product component', () => {
    const { getByText, baseElement } = render(<MockedNavigator component={ProductComponent} />);
    expect(getByText('Buzz')).not.toBeNull();
    expect(getByText('Alcohol by volume: ')).not.toBeNull();
    expect(getByText('4.5')).not.toBeNull();
    expect(getByText('Ibu: ')).not.toBeNull();
    expect(getByText('4.1')).not.toBeNull();
    expect(baseElement).toMatchSnapshot();
  });
  it ('Should click on the product', () => {
    const { getByTestId, getByText } = render(<MockedNavigator component={ProductComponent} />);
    const button = getByTestId('product');
    fireEvent(button, 'press');
  })
});