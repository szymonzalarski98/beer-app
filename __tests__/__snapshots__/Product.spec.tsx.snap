// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Product component Should render product component 1`] = `undefined`;
